src/app/server/Mdns.h

```diff
@@ -24,21 +24,31 @@ namespace chip {
 namespace app {
 namespace Mdns {
 
+enum class CommissioningMode
+{
+    kDisabled,                        // Commissioning Mode is disabled, CM=0, AC=0 in DNS-SD key/value pairs
+    kEnabled,                         // Basic Commissioning Mode, CM=1, AC=0 in DNS-SD key/value pairs
+    kEnabledAsAdditionalCommissioning // Enhanced Commissioning Mode, CM=1, AC=1 in DNS-SD key/value pairs
+};
+
 /// Start operational advertising
 CHIP_ERROR AdvertiseOperational();
 
 /// Set MDNS commissioner advertisement
 CHIP_ERROR AdvertiseCommissioner();
 
 /// Set MDNS commissionable node advertisement
-CHIP_ERROR AdvertiseCommissionableNode();
+CHIP_ERROR AdvertiseCommissionableNode(CommissioningMode mode);
 
-/// Set MDNS advertisement
-// CHIP_ERROR Advertise(chip::Mdns::CommssionAdvertiseMode mode);
-CHIP_ERROR Advertise(bool commissionableNode);
+/// (Re-)starts the minmdns server using default state
+/// - if device has not yet been commissioned, then commissioning mode will show as enabled (CM=1, AC=0)
+/// - if devica has been commissioned, then commissioning mode will show as disabled (CM=0, AC=0)
+void StartServer();
 
 /// (Re-)starts the minmdns server
-void StartServer();
+/// - if device has not yet been commissioned, then commissioning mode will show as enabled (CM=1, AC=0)
+/// - if devica has been commissioned, then commissioning mode will reflect the state of mode argument
+void StartServer(CommissioningMode mode);
```

Comment:
Why do we have two StartServer commands? if the default is CM=0, AC=0, why not setting mode to the default of the startserver without arguments? that  would make the header clearer. Just keeping:

```cpp
void StartServer(CommisioningMode mode = CommissioningMode::kDisabled);
```

Reply:
fixed